### ISSUE TITLE
The Analysis resolution is stuck at 1072*1072

### ISSUE THREAD

**Issue by dorrin-sot** [Apparence-io/CamerAwesome#301]
## Steps to Reproduce

```
CameraAwesomeBuilder.awesome(
        aspectRatio: CameraAspectRatios.ratio_1_1,
        sensor: Sensors.front,
        onImageForAnalysis: (img) => _analyzeImage(img),
        imageAnalysisConfig: AnalysisConfig(
          outputFormat: InputAnalysisImageFormat.nv21,
          width: 4000,
          maxFramesPerSecond: 20,
        ),
        ...
)
```

## Expected results

The **Analysis** resolution (the image received in `onImageForAnalysis`) would be my phone's max resolution which is 1440*2560

## Actual results

The **Analysis** resolution is 1072*1072

## About your device

| Brand   | Model       | OS        |
| ------- | ----------- | --------- |
| *Samsung* | *Galaxy A03*  | *Android 13*  |

## Your flutter version

Run this in your command line 
```flutter --version```

```
Flutter 3.7.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 9944297138 (7 weeks ago) • 2023-02-08 15:46:04 -0800
Engine • revision 248290d6d5
Tools • Dart 2.19.2 • DevTools 2.20.1
```
</details>

**Comment by apalala-dev** [comment]
On Android, we rely on the CameraX API and its final resolution may vary based on device capabilities and use cases (wether you want to record video or take photo in addition to image analysis).
Setting a target width will only try to find a close resolution that fits the rest of your configuration, which may explain this limitation.

Why would you want such an high resolution image for analysis?

**Comment by dorrin-sot** [comment]
Because this is how `_analyzeImage` works
```
_analyzeImage(AnalysisImage img) {
  final face = extract face data from img
  if (some condition for face that needs to be exact) {
    save picture
  }
}
```
Now when I want to save the picture the quality is too low.

I also tried using `CameraState` to capture the image right then but it has a few seconds of delay and so the condition may not be true after it's done.

**Comment by dorrin-sot** [comment]
Forgot to mention that I also need to do some Image processing that includes cropping the image. Which reduces the quality further so I need all the pixels I can get :)

**Comment by apalala-dev** [comment]
Having a huge quality in the image analysis processing will probably make it slow (at least on lower end devices).

You could eventually try to make a screenshot of the widget when you detect that your condition is met, but that would depend on the screen resolution.

If you lower the resolution (and eventually the maxFps), does taking a photo still takes that much time?

**Comment by dorrin-sot** [comment]
Still takes a few seconds :(

**Comment by apalala-dev** [comment]
That's weird, your phone might be too low-end...

What resolution would be acceptable for the image analysis? I'm trying something on an other branch that might help with your use case

**Comment by dorrin-sot** [comment]
As high as possible :) (2160*3840 would be ideal)

**Comment by dorrin-sot** [comment]
Another thing is I just logged the size of the input AnalysisImage in `_analyzeImage` and found out that it is 720*720 despite being set to 100 in AnalysisConfig.

**Comment by dorrin-sot** [comment]
But I have tried using a higher end phone that does normally have 4K resolution but its analysis image resolution was still around 1080*1080

**Comment by g-apparence** [comment]
It's not because your phone has a 4K resolution that analysis will allow you to analyze images with such a high quality. 
Analysis is aimed for analyzing content using AI, not taking great pictures. 
AI doesn't require such high quality images. (The more pixel the more power it requires). 

What do you want to do with such big analysis images ?

**Comment by dorrin-sot** [comment]
Another way that might work is raise the quality to max and cache that image, then pass a reduced version of that to the Face Analyzer.

**Comment by urossm** [comment]
I had the same issues with the resolutions, it was only giving me 1:1 aspect ratios, for example my phone, OnePlus Nord, has a maximum image size of 4000x3000 and when I create CameraAwesomeBuilder.previewOnly widget and no matter what image width and aspect ratio I set in imageAnalysisConfig, it always gives me 1:1 ratio image. For example if I set image width to 4000 and aspect ratio to 4:3 it should give me 4000x3000 image since that's what my phone supports but it gives me 3000x3000, same if I put let's say image width to 2000, it gives me 2558x2558 image. 

So I started messing with the native files in the CamerAwesome and I found a bug that solves this issue. in android>src.main>kotlin.com.apparance>camerawesome>cameraX>ImageAnalisysBuilder.kt on line 66 where it builds ImageAnalysis, it uses function setTargetResolution(Size(width, height)) but what fixes this issue is if you switch width and height and write Size(height, width) it works as it should. After that in flutter if I set image width to 4000 in imageAnalysisConfig it gives me 4000x3000 image and if I switch aspect ratio to 16:9 and set image width to 3840 it gives me image with size of 3840x2160 as it should. 

I don't know if that's bug with cameraX where it internally switches width and height or is something related with CamerAwesome but I hope this will be tested and implemented in some other versions because other than that, the plugin is awesome

**Comment by spydr1** [comment]
I also have same issue and i solved @urossm's solution. 
Thank you so much for your solution.

**Comment by DobbliDeveloper** [comment]
This problem is still not solved in version 2.0.1

**Comment by Nuyoahhjh** [comment]
> I had the same issues with the resolutions, it was only giving me 1:1 aspect ratios, for example my phone, OnePlus Nord, has a maximum image size of 4000x3000 and when I create CameraAwesomeBuilder.previewOnly widget and no matter what image width and aspect ratio I set in imageAnalysisConfig, it always gives me 1:1 ratio image. For example if I set image width to 4000 and aspect ratio to 4:3 it should give me 4000x3000 image since that's what my phone supports but it gives me 3000x3000, same if I put let's say image width to 2000, it gives me 2558x2558 image.
> 
> So I started messing with the native files in the CamerAwesome and I found a bug that solves this issue. In android>src.main>kotlin.com.apparence>camerawesome>cameraX>ImageAnalisysBuilder.kt on line 66 where it builds ImageAnalysis, it uses function setTargetResolution(Size(width, height)) but what fixes this issue is if you switch width and height and write Size(height, width) it works as it should. After that in flutter if I set image width to 4000 in imageAnalysisConfig it gives me 4000x3000 image as it should have in the first place and if I switch aspect ratio to 16:9 and set image width to 3840 it gives me image with size of 3840x2160 as it should.
> 
> I don't know if that's bug with CameraX where it internally switches width and height or is something related with CamerAwesome but I hope this will be tested and implemented in some other versions because other than that, the plugin is awesome

Thank you. I have the same question.
